taichi/python/export_lang.cpp

```diff
@@ -866,10 +871,16 @@ void export_lang(py::module &m) {
       .def(py::init<>())
       .def("create_root", &SNodeRegistry::create_root,
            py::return_value_policy::reference);
-  m.def("finalize_snode_tree",
-        [](SNodeRegistry *registry, const SNode *root, Program *program) {
-          program->add_snode_tree(registry->finalize(root));
-        });
+  m.def(
+      "finalize_snode_tree",
+      [](SNodeRegistry *registry, const SNode *root,
+         Program *program) -> SNodeTree * {
+        return program->add_snode_tree(registry->finalize(root));
+      },
+      py::return_value_policy::reference);
+  m.def("destroy_snode_tree", [](const int id, Program *program) {
+    program->destroy_snode_tree(id);
```

Comment:
```suggestion
  m.def("destroy_snode_tree", [](SNodeTree *ptr, Program *program) {
    program->destroy_snode_tree(id);
```
Pass in a `SNodeTree` pointer?